oauth2_http/javatests/com/google/auth/oauth2/CertificateIdentityPoolSubjectTokenSupplierTest.java

```diff
@@ -143,4 +157,274 @@ public void getSubjectToken_success() throws Exception {
+
+    CertificateFactory cf = CertificateFactory.getInstance("X.509");
+
+    // Parse and encode the leaf certificate.
+    X509Certificate expectedLeafCert =
+        (X509Certificate) cf.generateCertificate(new ByteArrayInputStream(testCertBytesFromFile));
+    String expectedEncodedLeaf = Base64.getEncoder().encodeToString(expectedLeafCert.getEncoded());
+
+    // Parse and encode the intermediate certificate.
+    X509Certificate expectedIntermediateCert =
+        (X509Certificate)
+            cf.generateCertificate(new ByteArrayInputStream(intermediateCertBytesFromFile));
+    String expectedEncodedIntermediate =
+        Base64.getEncoder().encodeToString(expectedIntermediateCert.getEncoded());
+
+    // Expected: JSON array with encoded leaf and then encoded intermediate.
+    String[] expectedCertChainArray =
+        new String[] {expectedEncodedLeaf, expectedEncodedIntermediate};
+    String expectedSubjectToken = OAuth2Utils.JSON_FACTORY.toString(expectedCertChainArray);
+
+    // Execute
+    String actualSubjectToken = supplier.getSubjectToken(mockContext);
+
+    // Verify
+    assertEquals(expectedSubjectToken, actualSubjectToken);
+  }
+
+  @Test
+  public void getSubjectToken_trustChainWrongOrder_throwsIllegalArgumentException() {
+    ClassLoader classLoader = getClass().getClassLoader();
+    URL trustChainUrl = classLoader.getResource("trust_chain_wrong_order.pem");
+    assertNotNull("Test trust chain file (wrong order) not found!", trustChainUrl);
+    String trustChainPath = new File(trustChainUrl.getFile()).getAbsolutePath();
+    when(mockCertificateConfig.getTrustChainPath()).thenReturn(trustChainPath);
+
+    String expectedRootErrorMessage =
+        "The leaf certificate should only appear at the beginning of the trust chain file, or be omitted entirely.";
+    String expectedOuterExceptionMessage =
+        "Trust chain misconfiguration: " + expectedRootErrorMessage;
+
+    // Execute & Verify
+    IOException exception =
+        assertThrows(IOException.class, () -> supplier.getSubjectToken(mockContext));
+
+    assertEquals(expectedOuterExceptionMessage, exception.getMessage());
+
+    Throwable cause = exception.getCause();
+    assertNotNull("Exception cause should not be null", cause);
+    assertTrue(
+        "Exception cause should be an IllegalArgumentException",
+        cause instanceof IllegalArgumentException);
+    assertEquals(expectedRootErrorMessage, cause.getMessage());
+  }
+
+  @Test
+  public void getSubjectToken_trustChainOnlyLeaf_success() throws Exception {
+    // Configure mock to use the leaf certificate file itself as the trust chain file,
+    // simulating a scenario where the trust chain file contains only the leaf.
+    ClassLoader classLoader = getClass().getClassLoader();
+    URL trustChainUrl = classLoader.getResource("x509_leaf_certificate.pem");
+    assertNotNull("Leaf certificate file not found!", trustChainUrl);
```

Comment:
This should be trustchain file not found, right? In this case, the trustchain only contains the leaf cert.

Reply:
The certificate file is being used for both leaf and trust chain here. I've updated the message to `Test resource 'x509_leaf_certificate.pem' (used as trust chain) not found!` to be a bit more clear.